app/uk/gov/hmrc/helptosavefrontend/config/frontendGlobal.scala

```diff
@@ -70,3 +78,26 @@ object AuditFilter extends FrontendAuditFilter with RunMode with AppName with Mi
 
   override def controllerNeedsAuditing(controllerName: String) = ControllerConfiguration.paramsForController(controllerName).needsAuditing
 }
+
+/**
+  * Checks if the request has cookie with a HTS session ID in it - if not, the filter creates one and
+  * performs `whenNoSession` with the new session ID in it
+  */
+@Singleton
+class SessionFilter[A](whenNoSession: => Result)(implicit app: Application) extends EssentialFilter {
+
+  implicit val ex = app.injector.instanceOf[ExecutionContext]
+  val sessionIdKey = app.configuration.underlying.getString("microservice.services.keystore.session-key")
+
+  private def createHtsCookie() = Cookie(name = sessionIdKey, value = s"hts-session-${UUID.randomUUID}")
```

Comment:
The `hts-session` prefix in the cookie value seems redundant given that the name already qualifies it